src/components/avatar-creator-raw.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { FC, useEffect, useRef } from 'react';
+import { AvatarCreatorConfig, IFrameEvent } from '../types';
+import { JSONTryParse } from '../utils';
+import { AvatarCreatorEvent } from '../events';
+import { useAvatarCreatorUrl } from '../hooks/use-avatar-creator-url';
+
+const MESSAGE_EVENT = 'message';
+const RPM_TARGET = 'readyplayerme';
+const IFRAME_READY_EVENT = 'v1.frame.ready';
+
+export type AvatarCreatorRawProps = {
+  subdomain: string;
+  className?: string;
+  config?: AvatarCreatorConfig;
+};
+
+export type EventReceivedProps = {
+  onEventReceived?: (event: IFrameEvent<AvatarCreatorEvent>) => void;
+};
+
+/**
+ * AvatarCreatorRaw is a React component that allows you to create an avatar using Ready Player Me and receive avatar URL. It exposes the raw events in one callback to allow you to write more custom logic around the event handling.
+ * @param subdomain The subdomain of your Ready Player Me instance.
+ * @param className The css styles to apply to this iframe.
+ * @param avatarCreatorConfig The configuration for the AvatarCreator component.
+ * @param onEventReceived A callback that is called when an avatar creator event is received.
+ * @returns A React component.
+ */
+export const AvatarCreatorRaw: FC<AvatarCreatorRawProps & EventReceivedProps> = ({ subdomain, className, config, onEventReceived }) => {
+  const iframeRef = useRef<HTMLIFrameElement | null>(null);
+  const url = useAvatarCreatorUrl(subdomain, config);
+
+  const subscribeToAvatarCreatorEvents = () => {
+    if (!iframeRef.current?.contentWindow) return;
+
+    iframeRef.current?.contentWindow?.postMessage(
+      JSON.stringify({
+        target: RPM_TARGET,
+        type: 'subscribe',
+        eventName: 'v1.**',
+      }),
+      '*'
+    );
+  };
+
+  const subscribe = (event: any) => () => {
+    const iframeEvent = JSONTryParse<IFrameEvent<AvatarCreatorEvent>>(event);
+
+    if (iframeEvent?.source !== RPM_TARGET) return;
+
+    if (iframeEvent?.eventName === IFRAME_READY_EVENT) {
+      subscribeToAvatarCreatorEvents();
+      return;
+    }
+
+    onEventReceived?.(iframeEvent);
+  };
+
+  useEffect(() => {
+    window.addEventListener(MESSAGE_EVENT, subscribe);
+
+    return () => {
+      window.removeEventListener(MESSAGE_EVENT, subscribe);
+    };
+  }, []);
+
+  return <iframe title="Ready Player Me" ref={iframeRef} src={url} className={className} allow="camera *; clipboard-write" />;
```

Comment:
Regarding styling the iframe, I think we should have the setting in styles set so that it has already  max width and height based of the parent component.

Reply:
Seems like an extra step of defining the style for it.

Reply:
fallback option as [this](https://github.com/readyplayerme/rpm-react-sdk/blob/master/src/components/avatar-creator.tsx#L15C1-L19C3):

I actually did the same style when I first tested this out, because the initial IFrame was quite out of place.

Reply:
It's not full screen. I'd say it's rather container size on whereever you want to add it. :)

Reply:
That's also potentially weird/unexpected behavior which wouldn't be ideal 😅

Reply:
I don't see a reason why would we want to override the style of the iframe at all. The content itself is anyways out of the reach and if they want to style it , they can just wrap it into the separate div.

Reply:
Put a different way, why would we not give developers access to style the iframe itself, what are the actual downsides to doing that? Other than the fact, that for 1 specific use-case a developer has to add a bit of CSS when they otherwise wouldn't?